src/System.IO.Ports/src/System/IO/Ports/InternalResources.cs

```diff
@@ -77,7 +77,10 @@ internal static void WinIOError(int errorCode, String str)
                         throw new UnauthorizedAccessException(string.Format(SR.UnauthorizedAccess_IODenied_Path, str));
 
                 case Interop.Errors.ERROR_FILENAME_EXCED_RANGE:
-                    throw new PathTooLongException(SR.IO_PathTooLong);
+                    if (string.IsNullOrEmpty(str))
+                        throw new PathTooLongException(SR.IO_PathTooLong);
+                    else
+                        throw new PathTooLongException(SR.Format(SR.IO_PathTooLong_Path, str));
```

Comment:
Nit: it'd be nice to use braces here on both the if/else blocks, or else condense this into a single throw, e.g.
```C#
throw string.IsNullOrEmpty(str) ?
    new PathTooLongException(SR.IO_PathTooLong) :
    new PathTooLongException(SR.Format(SR.IO_PathTooLong_Path, str));
```

Reply:
(though I see now that it's just copying the style used elsewhere in the file, so ok)